pycqed/instrument_drivers/physical_instruments/QuTech_ControlBoxdriver.py

```diff
@@ -94,7 +94,7 @@ def add_params(self):
                 label='Integraion weights input {}'.format(i),
                 get_cmd=self._wrap_ch_get_fun(self._get_int_weights, i),
                 set_cmd=self._wrap_ch_set_fun(self._set_int_weights, i),
-                vals=vals.Anything())
+                vals=vals.Ints(-128, 127))
```

Comment:
This will not work. It will check if the weights is an Integer where it should be an Array. 

Try using the Arrays validator (this allows both specifying a min and maxvalue as well as a shape. Maybe you can even make a custom one that also checks type. 

There is also a good reason to use validators as they have auto generated docstrings so that the user can easily see what input is expected.